File: app/src/main/java/ru/geekbrains/notes/view/NoteFragment.java
package ru.geekbrains.notes.view;

import android.graphics.Typeface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.fragment.app.Fragment;

import ru.geekbrains.notes.Note;
import ru.geekbrains.notes.R;

public class NoteFragment extends Fragment {

    public static String KEY_NOTE = "note";
    private Note note;

    public static NoteFragment newInstance(Note note) {
        NoteFragment fragment = new NoteFragment();
        Bundle bundle = new Bundle();
        bundle.putParcelable(KEY_NOTE, note);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_note, container, false);
        TextView titleView = view.findViewById(R.id.title_view);
        TextView textView = view.findViewById(R.id.text_view);
        titleView.setText(this.note.getTitle());
        textView.setText(this.note.getText());
        titleView.setTypeface(null, Typeface.BOLD);
        textView.setTypeface(null, Typeface.ITALIC);
        return view;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if(getArguments() != null)
            this.note = getArguments().getParcelable(KEY_NOTE);
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        outState.putParcelable(KEY_NOTE, this.note);
        super.onSaveInstanceState(outState);
    }
}

File: app/src/main/java/ru/geekbrains/notes/view/NoteFragment.kt
package ru.geekbrains.notes.view

import android.graphics.Typeface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import ru.geekbrains.notes.model.Note
import ru.geekbrains.notes.R

class NoteFragment : Fragment() {
    private var note = Note()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return initView(inflater, container)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null)
            requireArguments().getParcelable<Note>(KEY_NOTE)?.let { note = it }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        outState.putParcelable(KEY_NOTE, note)
        super.onSaveInstanceState(outState)
    }

    private fun initView(
        inflater: LayoutInflater,
        container: ViewGroup?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_note, container, false)
        val titleView = view.findViewById<TextView>(R.id.title_view)
        val textView = view.findViewById<TextView>(R.id.text_view)
        titleView.text = note.title
        textView.text = note.text
        titleView.setTypeface(null, Typeface.BOLD)
        textView.setTypeface(null, Typeface.ITALIC)
        return view
    }

    companion object {
        var KEY_NOTE = "note"
        fun newInstance(note: Note): NoteFragment {
            val fragment = NoteFragment()
            val bundle = Bundle()
            bundle.putParcelable(KEY_NOTE, note)
            fragment.arguments = bundle
            return fragment
        }
    }
}
